src/Product/Unit/Loader.php

```diff
@@ -228,21 +229,22 @@ private function _buildQuery($revisionID = null)
 			->join('product', 'product_unit.product_id = product.product_id') // Join product table to filter out units where product is hard deleted
 			;
 
-			if (null !== $getRevision) {
-				$this->_queryBuilder
-					->leftJoin('product_unit_info', '
-						product_unit_info.unit_id = product_unit.unit_id AND
-						revision_id = (:revisionID?q)
-					')
-					->addParams(['revisionID' => $getRevision]);
-			} else {
-				$this->_queryBuilder
-					->leftJoin('product_unit_info', '
-						product_unit_info.unit_id = product_unit.unit_id AND
-						revision_id = 1
-					')
-				;
-			}
+		if (is_numeric($getRevision)) {
```

Comment:
The statements have reversed. If the revision number is a number it gets used (rather than using an odd subquery) otherwise the new query to get the latest is used.